### ISSUE TITLE
Inconsitency in model instance populated vs not populated

### ISSUE THREAD

**Issue by delgiudices** [Automattic/mongoose#7914]
I've been bitten by this one a few times since I started using mongoose, either I need a mindset or this is a nice feature request.

Say I declare a model like following:

```
const Make = mongoose.model('Make', new mongoose.Schema({
  name: String,
});

const Car = mongoose.model('Person', new mongoose.Schema({
  name: String,
  make: { type: mongoose.Schema.Types.ObjectId, ref: 'Make' },
});
```

And I have the following function:

```
function resolveCar(carInstance) {
  return {
    name: carInstance.name,
    makeName: carInstance.make.name
  };
}
```

If i run:

```
resolveCar(await Car.findById('<car_id>').populate('make'))
```

Everything works as expected, however if I run:

```
resolveCar(await Car.findById('<car_id>'))
```

Function will not resolve correctly because it is expecting a car instance where make is an object containing the make instance, this discrepancy of instance states depending on where it was called with populate or not makes it tricky to write functions that depend on model instances.

Ideally we could have a instance method. e.g. `car.getMake()` that automatically resolves to the make instance if it's loaded or fetches it from the DB if it's not.

**Comment by vkarpov15** [comment]
I agree this gets confusing. Right now the best approach is to use [`doc.populated()` to check whether the path is populated or not](https://mongoosejs.com/docs/api/document.html#document_Document-populated), and `populate()` if it isn't.

```javascript
if (!car.populated('make')) {
  await car.populate('make').execPopulate();
}
```

We haven't been able to think of a cleaner way yet, so I'll keep this open for now as something to think about.

**Comment by vkarpov15** [comment]
I'm going to close this one since it is fairly easy to implement:

```javascript
carSchema.methods.getMake = async function() {
  if (!this.populated('make')) {
    await this.populate('make').execPopulate();
  }
  return this.make;
};
```

You could also have a `makeId` and then have `make` as a [populated virtual](https://mongoosejs.com/docs/tutorials/virtuals.html#populate).
